### ISSUE TITLE
(sp_Blitz) Add @MaxHistoryReport parameter for Errors Logged Recently in the Default Trace

### ISSUE THREAD

**Issue by DBAKen** [BrentOzarULTD/SQL-Server-First-Responder-Kit#1524]
I would like to add the parameter @MaxHistoryReport to sp_Blitz in order to keep from seeing the same error reported everyday for 30 days.  This is a great script and I have it scheduled to run on every SQL Server I manage every morning and then I get an aggregated report emailed to me.

The stored procedure declaration would look like:
	@MaxHistoryReport INT = 30,

The code under /* Reliability - Errors Logged Recently in the Default Trace */ would change (in 2 places)
From: 
AND t.StartTime > DATEADD(dd, -30, GETDATE())

To:
AND t.StartTime > DATEADD(dd, (@MaxHistoryReport * -1), GETDATE())

**Comment by BrentOzar** [comment]
Ooo, that's interesting. Things to think about:

- This affects a LOT of existing checks (backups, checkdb, etc). You would want to know which checks were affected, and which ones weren't. If you want to do this, you'll need to make a list of the existing checks in sp_Blitz along with their current number of days back to check, and whether or not the new parameter will modify that.
- I'd name the parameter @DaysBackToCheck so that it's more self-explanatory.
- Make it a tinyint (because somebody's gonna pass a giant value in, and it's going to break stuff, and this way you don't have to worry about positives/negatives either.)
- Give it a default value (and I'm sure there's going to be a ton of community discussion around that value if it affects things like backups and checkdb.)

Are you sure you want to tackle something this big? (I definitely wouldn't.) If you do, I'm all for you doing it - the next step would be to build that list of existing checks.

**Comment by DBAKen** [comment]
Yeah, I started looking at everywhere DATEADD is used and found some -7, -14 and -30 day values.  BTW, I am using the 20180101 version.   I started to just change it in my copy and then thought that I then have to keep track so I change it in the next version I implement.  I understand that there are different date sensitivities for different activities.  I guess I could post-process the results before I email my report.

Anyway, let me take a stab at compiling the list and see what it looks like.  Thanks for the feedback.  Keep up the good work you do.

**Comment by BrentOzar** [comment]
You bet! Yeah, whatever you do, don't try to work on this without installing Github and following the directions there. Your code will drift out of sync fast - we do checkins on this every couple/few weeks. Here's the contributing guide:

https://github.com/BrentOzarULTD/SQL-Server-First-Responder-Kit/blob/dev/CONTRIBUTING.md

I'm going to assign the issue to you, but that's just so folks know you're working on it. If you give up, just report back here and we'll close it. (We'll also close it if you don't make progress within a couple of weeks - we tend to not keep issues open if they're not moving forward.)

Thanks sir!

**Comment by BrentOzar** [comment]
Yeah, makes sense. I love the idea, but it's a ton of work. Thanks anyway!
